packages/mdc-tab/index.ts

```diff
@@ -0,0 +1,143 @@
+  private content_!: HTMLElement; // assigned in initialize();
+  private handleClick_!: EventListener; // assigned in initialize();
+
+  initialize(
+      rippleFactory: RippleFactory = (el, foundation) => new MDCRipple(el, foundation),
+      tabIndicatorFactory: TabIndicatorFactory = (el) => new MDCTabIndicator(el),
+  ) {
+    this.id = this.root_.id;
+    const rippleSurface = this.root_.querySelector<HTMLElement>(MDCTabFoundation.strings.RIPPLE_SELECTOR)!;
+    const rippleAdapter = {
+      ...MDCRipple.createAdapter(this),
+      addClass: (className: string) => rippleSurface.classList.add(className),
+      removeClass: (className: string) => rippleSurface.classList.remove(className),
+      updateCssVariable: (varName: string, value: string) => rippleSurface.style.setProperty(varName, value),
+    };
+    const rippleFoundation = new MDCRippleFoundation(rippleAdapter);
+    this.ripple_ = rippleFactory(this.root_, rippleFoundation);
+
+    const tabIndicatorElement = this.root_.querySelector(MDCTabFoundation.strings.TAB_INDICATOR_SELECTOR)!;
+    this.tabIndicator_ = tabIndicatorFactory(tabIndicatorElement);
+    this.content_ = this.root_.querySelector<HTMLElement>(MDCTabFoundation.strings.CONTENT_SELECTOR)!;
+  }
+
+  initialSyncWithDOM() {
+    this.handleClick_ = this.foundation_.handleClick.bind(this.foundation_);
+    this.listen('click', this.handleClick_);
+  }
+
+  destroy() {
+    this.unlisten('click', this.handleClick_);
+    this.ripple_.destroy();
+    super.destroy();
+  }
+
+  getDefaultFoundation(): MDCTabFoundation {
+    // tslint:disable:object-literal-sort-keys
+    return new MDCTabFoundation({
+      setAttr: (attr, value) => this.root_.setAttribute(attr, value),
+      addClass: (className) => this.root_.classList.add(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+      hasClass: (className) => this.root_.classList.contains(className),
+      activateIndicator: (previousIndicatorClientRect) => this.tabIndicator_.activate(previousIndicatorClientRect),
+      deactivateIndicator: () => this.tabIndicator_.deactivate(),
+      notifyInteracted: () => this.emit<TabInteractionEventDetail>(
+        MDCTabFoundation.strings.INTERACTED_EVENT, {tabId: this.id}, true /* bubble */),
+      getOffsetLeft: () => this.root_.offsetLeft,
+      getOffsetWidth: () => this.root_.offsetWidth,
+      getContentOffsetLeft: () => this.content_.offsetLeft,
+      getContentOffsetWidth: () => this.content_.offsetWidth,
+      focus: () => this.root_.focus(),
+    });
+    // tslint:enable:object-literal-sort-keys
+  }
+
+  /**
+   * Getter for the active state of the tab
+   */
+  get active(): boolean {
+    return this.foundation_.isActive();
+  }
+
+  set focusOnActivate(focusOnActivate: boolean) {
+    this.foundation_.setFocusOnActivate(focusOnActivate);
+  }
+
+  /**
+   * Activates the tab
+   */
+  activate(computeIndicatorClientRect?: ClientRect | null) {
```

Comment:
in tab-indicator we changed this to just `computeIndicatorClientRect?: ClientRect`

Reply:
Done